### PR TITLE
Create esp32spi_tcp_client.py

### PR DESCRIPTION
**Pull request by anecdata**
A simple TCP client example using ESP32SPI.

Requires running a TCP server listening on the same port. For example, CPython on a general-purpose computer:
https://github.com/adafruit/circuitpython/blob/main/tests/circuitpython-manual/socketpool/client/host-server.py
or CircuitPython native sockets:
https://github.com/adafruit/circuitpython/blob/main/tests/circuitpython-manual/socketpool/server/cpy-server.py

### CODE REVIEW

File: examples/esp32spi_tcp_client.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+from digitalio import DigitalInOut
+from adafruit_esp32spi import adafruit_esp32spi
+import adafruit_esp32spi.adafruit_esp32spi_socket as socket
+import adafruit_requests as requests
+from secrets import secrets
+
+
+TIMEOUT = 5
+# adjust host and port to match server
+HOST = "192.168.10.179"
```
Comment (brentru) [inline examples/esp32spi_tcp_client.py:14]: Might want to add a different IP or placeholder text/IP here?

File: examples/esp32spi_tcp_client.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+PORT = 5000
+
+# PyPortal or similar; adjust pins as needed
+spi = board.SPI()
```
Comment (brentru) [inline examples/esp32spi_tcp_client.py:18]: Same here, or pull from secrets

File: examples/esp32spi_tcp_client.py (hunk 1 above)
Comment (anecdata) [reply examples/esp32spi_tcp_client.py:14]: different IP? I'm happy to dummy it up somehow, not sure what you're thinking?

File: examples/esp32spi_tcp_client.py (hunk 2 above)
Comment (anecdata) [reply examples/esp32spi_tcp_client.py:18]: I agree secrets would be a good approach in general, but it isn't standard for host and port, so would require explanatory text on how to do that... so probably same approach as above

File: examples/esp32spi_tcp_client.py (hunk 1 above)
Comment (anecdata) [reply examples/esp32spi_tcp_client.py:14]: I could use wifitest.adafruit.com and port 80 ...that would give some 404 rexponse or similar

Comment (anecdata) [reply examples/esp32spi_tcp_client.py:14]: `Receiving
b'HTTP/1.1 400 Bad Request\r\nServer: nginx/1.10.3 (Ubuntu)\r\nDate: Mon, 15 Mar 2021 16:24:42 GMT\r\nContent-Type: text/html\r\nContent-L'`

**Review by brentru — APPROVED** [review]

**Pull request merged**